modules/nw-ipfailover-configuring-check-notify-scripts.adoc

```diff
@@ -0,0 +1,106 @@
+
+Each IP failover pod manages a Keepalived daemon that manages one or more virtual IPs (VIP) on the node where the pod is running. The Keepalived daemon keeps the state of each VIP for that node. A particular VIP on a particular node may be in `master`, `backup`, or `fault` state.
+
+When the check script for that VIP on the node that is in `master` state fails, the VIP on that node enters the `fault` state, which triggers a renegotiation. During renegotiation, all VIPs on a node that are not in the `fault` state participate in deciding which node takes over the VIP. Ultimately, the VIP enters the `master` state on some node, and the VIP stays in the `backup` state on the other nodes.
+
+When a node with a VIP in `backup` state fails, the VIP on that node enters the `fault` state. When the check script passes again for a VIP on a node in the `fault` state, the VIP on that node exits the `fault` state and negotiates to enter the `master` state. The VIP on that node may then enter either the `master` or the `backup` state.
+
+As cluster administrator, you can provide an optional notify script, which is called whenever the state changes. Keepalived passes the following three parameters to the script:
+
+* `$1` - `group` or `instance`
+* `$2` - Name of the `group` or `instance`
+* `$3` - The new state: `master`, `backup`, or `fault`
+
+The check and notify scripts run in the IP failover pod and use the pod file system, not the host file system. However, the IP failover pod makes the host file system available under the `/hosts` mount path. When configuring a check or notify script, you must provide the full path to the script. The recommended approach for providing the scripts is to use a config map.
+
+The full path names of the check and notify scripts are added to the Keepalived configuration file, `_/etc/keepalived/keepalived.conf`, which is loaded every time Keepalived starts. The scripts can be added to the pod with a config map as follows.
+
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create the desired script and create a config map to hold it. The script has no input arguments and must return `0` for `OK` and `1` for `fail`.
++
+The check script, `_mycheckscript.sh_`:
++
+[source,bash]
+----
+#!/bin/bash
+    # Whatever tests are needed
+    # E.g., send request and verify response
+exit 0
+----
+
+. Create the config map:
++
+[source,terminal]
+----
+$ oc create configmap mycustomcheck --from-file=mycheckscript.sh
+----
++
+. Add the script to the pod. The `defaultMode` for the mounted config map files must able to run is you use `oc` commands or edit the deployment configuration. A value of `0755`, `493` decimal, is typical:
++
+[source,terminal]
+----
+$ oc set env dc/ipfailover-keepalived \
```

Comment:
We created the ipfailover as Deployment instead of DeploymentConfig, so all `dc/ipfailover-keepalived` should be replaced by `deploy/ipfailover-keepalived`.

Reply:
Thank you! Updating.